dotCMS/src/test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentAsDraftActionletTest.java

```diff
@@ -0,0 +1,149 @@
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    @UseDataProvider("respectFrontendRolesValues")
+    public void shouldCallSaveDraftWithTheRightParameters(final TestCase testCase) throws DotSecurityException, DotDataException {
+        final WorkflowProcessor processor = mock(WorkflowProcessor.class);
+        final Map<String, WorkflowActionClassParameter> params = map();
+
+        final Contentlet contentlet = mock(Contentlet.class);
+        final User user  = mock(User.class);
+
+        when(processor.getUser()).thenReturn(user);
+        when(processor.getContentlet()).thenReturn(contentlet);
+
+        List<Category> categories = mock(List.class);
+
+        final List<Permission> permissions = mock(List.class);
+        when(this.permissionAPI.getPermissions(contentlet, false, true)).thenReturn(permissions);
+
+        final ContentletDependencies contentletDependencies = mock(ContentletDependencies.class);
+        when(processor.getContentletDependencies()).thenReturn(contentletDependencies);
+
+        final Contentlet contentletNew = mock(Contentlet.class);
+        when(contentletDependencies.getCategories()).thenReturn(categories);
+
+        when(processor.getContentletDependencies().isRespectAnonymousPermissions()).thenReturn(testCase.respectFrontendRoles);
+
+        when(this.contentletAPI.saveDraft(
+                contentlet, (ContentletRelationships) null, categories, permissions, user, testCase.respectFrontendRoles))
+                .thenReturn(contentletNew);
+
+        saveContentAsDraftActionlet.executeAction(processor, params);
+
+        //verify(processor, times(1)).setContentlet(contentletNew);
+        verify(this.contentletAPI, times(1)).saveDraft(
+                contentlet, (ContentletRelationships) null, categories, permissions, user, testCase.respectFrontendRoles
+        );
+
+        verify(this.categoryAPI, never()).getParents(contentlet, user, false);
+    }
+
+    /**
+     * When: {@link WorkflowProcessor#getContentletDependencies()} is null
+     * Should: call {@link ContentletAPI#saveDraft(Contentlet, Map, List, List, User, boolean)} with the right parameters
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    @UseDataProvider("respectFrontendRolesValues")
+    public void shouldCallSaveDraftWithTheRightParametersWhenContentletDependenciesIsNull(final TestCase testCase) throws DotSecurityException, DotDataException {
+        final WorkflowProcessor processor = mock(WorkflowProcessor.class);
+        final Map<String, WorkflowActionClassParameter> params = map();
+
+        final Contentlet contentlet = mock(Contentlet.class);
+        final User user  = mock(User.class);
+
+        when(processor.getUser()).thenReturn(user);
+        when(processor.getContentlet()).thenReturn(contentlet);
+
+        List<Category> categories = mock(List.class);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'categories' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337055)